Configura els sudoers

alexm opened the pull request:
Inicialment aquesta branca havia de configurar només els sudoers dels masovers però ha evolucionat cap a un rol propi que gestiona tota la configuració dels sudoers als servidors.

alexm committed:
Configura els sudoers de masovers

alexm committed:
Documenta el funcionament dels custom facts

alexm committed:
Generalitza per a la resta de sudoers

alexm committed:
Refactoritza per evitar duplicats

alexm committed:
Mou la plantilla de sudoers al nou rol

aniolm9 reviewed (approved):
LGTM

DPini commented on sudoers.yml:
```diff
@@ -0,0 +1,5 @@
+---
+- name: Sudoers role applies to all hosts
```
Ja que aquest rol aplica a tots els hosts, potser el podem fer que el rol base depengui de/requereixi de sudoers, de tal manera que passi a formar de la configuració base.

Entenc que potser la idea de deixar-ho a part es per a ser flexibles a l'hora d'afegir possibles nous hosts en un futur. Configuració de contenidors o similar per example.

DPini commented on roles/sudoers/tasks/main.yml:
```diff
@@ -0,0 +1,8 @@
+---
```
He vist que com a part del rol base, instalem sudo.

Una alternativa seria renombrar aquest rol a sudo, i fer que el rol sudo instal.li i configuri sudo i sudoers

Tampoc veig perquè no voldirem sudo a cap de les màquines administrades a Caliu, així que també trobo bé aquest approach.

NOTA: Potser estic massa influènciat per altres estructres de receptes de sistemes de configuració, principalment Salt Stack, i el que proposo no té sentit amb Ansible o no segueix good practices.

alexm committed:
Marca que el rol base depèn de sudoers

Com que el rol de sudoers aplica també a tots els servidors,
té tot el sentit del món que sigui una dependència del rol base.

Reference: https://github.com/caliu/ansible/pull/3#discussion_r615074528
Thanks: DPini

alexm committed:
Instal·la sudo al rol de sudoers

Reference: https://github.com/caliu/ansible/pull/3#discussion_r615076317
Thanks: DPini

alexm commented:
M'he fixat que tal com he declarat la dependència del rol base amb el de sudoers, primer s'aplica el de sudoers i no m'acaba de convèncer perquè preferiria que s'instal·li etckeeper i així començar a desar els canvis.

Se m'acut que podria crear un rol nou, que sigui una dependència també del base i que contingui les coses essencials com l'etckeeper i deixar al rol base les coses opcionals com ara utilitats generals, etc.

Què us sembla? Se us acut alguna manera millor?

alexm committed:
Mou la configuració essencial a un nou rol

D'aquesta manera el rol base es queda només amb les parts comuns
que no són essencials.

alexm commented:
Havia oblidat moure els defaults del sources.list a 6786ebf. Ho he resolt a 9b00dfb i he forçat els canvis.

alexm commented:
Segurament podríem considerar que el fail2ban també és essencial, però ara mateix prefereixo acabar aquesta proposta.

DPini commented:
LGTM! Gràcies Alex

aniolm9 reviewed (approved):
LGTM. Moltes gràcies per la feina, Alex!
